components/eam/src/physics/cam/cldera_passive_tracers.F90

```diff
@@ -0,0 +1,430 @@
+    do k = 1, pver
+       do i = 1, ncol
+       ! the explicit horizontal looping is not currently needed; maintained 
+       ! in case spatial dependence in e.g. AOA is ever desired 
+
+          ! ============ AOA ============
+          ! clock tracer with a source of 1 day/day everywhere above ~700hPa
+          if (pref_mid_norm(k) <= 0.7) then
+              ptend%q(i,k,ixaoa) = 1.0_r8 * aoa_scaling
+          else
+              ptend%q(i,k,ixaoa) = 0.0_r8
+              state%q(i,k,ixaoa) = 0.0_r8
+          end if
+
+          ! ============ E90 ============
+          ! dissipates with e-folding time of 90 days
+          ptend%q(i,k,ixe90) =  -(1/efold_e90) * state%q(i, k, ixe90)
+          ! apply surface flux to lowest model level
+          ! this mimics behavior of vertical_diffusion_tend()
+          ! last line adjusts for dry constituent
+          if(k == pver) then
+              ptend%q(i, k, ixe90) = ptend%q(i, k, ixe90) + &
+                                     gravit * state%rpdel(i, k) * sflx_e90 * & 
+                                     state%pdel(i,k) / state%pdeldry(i, k)
+          end if
+
+          ! ============ ST80_25 ============
+          ! dissipates with e-folding time of 25 days below ~80 hPa, 
+          ! constant concentration of 200 ppbv above
+          if (state%pmid(i, k) >= 8000._r8) then
+              ptend%q(i,k,ixst80) =  -(1/efold_st80) * state%q(i, k, ixst80)
+          else
+              ptend%q(i, k,ixst80) = 0._r8
+              state%q(i, k, ixst80) = 200e-9_r8  ! 200 ppbv to vmr
+          end if
+       end do
+    end do
+
+    ! -------------------- TRACER FLUXES --------------------
+    do i = 1, ncol
+       ! ====== AOA ======
+       ! no surface flux
+       cflx(i,ixaoa) = 0._r8
+
+       ! ====== E90 ======
+       ! surface flux handled in tendency computation above
+       cflx(i,ixe90) = 0._r8
+
+       ! ====== ST80_25 ======
+       ! no surface flux
+       cflx(i,ixst80) = 0._r8
+    end do
+           
+
+  end subroutine cldera_passive_tracers_timestep_tend
+
+!===========================================================================
+
+  subroutine init_cnst_3d(m, q, gcid)
+
+    use dyn_grid,    only : get_horiz_grid_d, get_horiz_grid_dim_d
+    use dycore,      only : dycore_is
+    use ref_pres,    only : pref_mid_norm
+
+    integer,  intent(in)  :: m       ! global constituent index
+    real(r8), intent(out) :: q(:,:)  ! kg tracer/kg dry air (gcol,plev)
+    integer,  intent(in)  :: gcid(:) ! global column id
+
+    real(r8), allocatable :: lat(:)
+    integer :: plon, plat, ngcols
+    integer :: j, k, gsize
+    !-----------------------------------------------------------------------
+    ! initialization below is on the DYNAMICS grid; length of gcol will 
+    ! be number of GLL points, not number of physics columns
+
+    if (masterproc) write(iulog,*) 'CLDERA_CLOCK CONSTITUENTS: INITIALIZING ',cnst_name(m),m
```

Comment:
Isn't labelling this `CLDERA_CLOCK CONSTITUENTS` misleading, since this now includes e90 and st80 which are NOT clock constituents?

Reply:
yes, this is a remnant from when this module only did AOA, will change